axelor-purchase/src/main/java/com/axelor/apps/purchase/service/ABCAnalysisServicePurchaseImpl.java

```diff
@@ -68,7 +71,11 @@ protected ABCAnalysisLine createABCAnalysisLine(ABCAnalysis abcAnalysis, Product
             JPA.clear();
         }
 
-        return setQtyWorth(abcAnalysisLineRepository.find(abcAnalysisLine.getId()), productQty, productWorth);
+        if(abcAnalysisLine != null){
+            setQtyWorth(abcAnalysisLineRepository.find(abcAnalysisLine.getId()), productQty, productWorth);
```

Comment:
Here we already call the find method, so no need to call it in setQtyWorth method.